src/openklant/management/commands/migrate_to_v2.py

```diff
@@ -0,0 +1,475 @@
+            soort_partij = self._get_soort_partij()
+        except KeyError:
+            logger.exception("Unable to determine soortPartij from subjectType")
+            return
+
+        data = dict(
+            indicatie_actief=True,
+            indicatie_geheimhouding=False,
+            rekeningnummers=[],
+            voorkeurs_rekeningnummer=None,
+            digitale_adressen=None,
+            voorkeurs_digitaal_adres=None,
+            soort_partij=soort_partij,
+            nummer=subject.nummer if subject else "",
+            partij_identificatie=subject.migrate() if subject else None,
+        )
+
+        if digitaal_adres:
+            data["digitale_adressen"] = [dict(uuid=digitaal_adres)]
+            data["voorkeurs_digitaal_adres"] = dict(uuid=digitaal_adres)
+
+        return Partij(**data)
+
+
+def _retrieve_klanten(url: str, access_token: str) -> Tuple[list[Klant], str | None]:
+    klanten_path = "/klanten/api/v1/klanten"
+
+    _url = urlparse(url)
+    _params = parse_qs(_url.query)
+
+    client = LegacyOpenKlantClient(f"{_url.scheme}://{_url.netloc}", access_token)
+    response_data = client.retrieve(_url.path or klanten_path, params=_params)
+
+    if not isinstance(response_data, dict):
+        logger.error(f"Unexpected response data returned: {response_data}")
+        return [], None
+
+    items = response_data.get("results", [])
+    klant_fields = {field.name: field for field in dataclass_fields(Klant)}
+    klanten = []
+
+    for data in items:
+        klant = Klant(
+            **{field: value for field, value in data.items() if field in klant_fields}
+        )
+
+        klanten.append(klant)
+
+    next_url = response_data.get("next")
+    return klanten, next_url
+
+
+def _save_klanten(url: str, klanten: list[Klant]) -> list[str]:
+    if not klanten:
+        logger.info("No klanten to save, moving on...")
+        return []
+
+    logger.info(f"Trying to create {len(klanten)} klanten through the V2 API")
+
+    digitaal_addressen_path = reverse(
+        "klantinteracties:digitaaladres-list",
+        kwargs=dict(version=settings.REST_FRAMEWORK["DEFAULT_VERSION"]),
+    )
+
+    partijen_path = reverse(
+        "klantinteracties:partij-list",
+        kwargs=dict(version=settings.REST_FRAMEWORK["DEFAULT_VERSION"]),
+    )
+
+    dummy_token = _generate_dummy_token()
+    openklant_client = OpenKlantClient(url, dummy_token)
+    generic_client = Client()
+    created_klanten = []
+
+    for klant in klanten:
+        digitaal_adres = klant.to_digitaal_adres()
+        digitaal_adres_ref = None
+
+        if digitaal_adres:
+            _data = openklant_client.create(
+                digitaal_addressen_path, digitaal_adres.dict()
+            )
+
+            if not isinstance(_data, dict):
+                logger.error(
+                    f"Unknown data received creating a digitaal adres: {_data}. "
+                    "Skipping klant."
+                )
+                continue
```

Comment:
I'm not sure what the user can actually do if some entries fail, but will it be visible in the logs which specific Klanten have failed so that they can be easily identified?